dask/array/tests/test_overlap.py

```diff
@@ -702,3 +723,53 @@ def test_trim_boundry(boundary):
     x_overlaped = da.overlap.overlap(x, 2, boundary=boundary)
     x_trimmed = da.overlap.trim_overlap(x_overlaped, 2, boundary=boundary)
     assert np.all(x == x_trimmed)
+
+
+def test_map_overlap_rechunks_array_if_needed():
+    # https://github.com/dask/dask/issues/6597
+    expected = np.arange(11)
+    x = da.from_array(expected, chunks=5)
+    y = x.map_overlap(lambda x: x, depth=2, boundary=0)
+    assert y.chunks == ((5, 4, 2),)
+    assert_array_equal(y, expected)
+
+
+def test_map_overlap_rechunks_array_along_multiple_dims_if_needed():
+    # https://github.com/dask/dask/issues/6688
+    rand = da.random.random((860, 1024, 1024), chunks=(1, 1024, 1024))
+    filtered = rand.map_overlap(
+        lambda arr: arr,
+        depth=(2, 2, 2),
+        boundary="reflect",
+    )
+    assert filtered.chunks == ((2,) * 430, (1024,), (1024,))
+
+
+@pytest.mark.parametrize(
+    "chunks,expected",
+    [
+        [(10,), (10,)],
+        [(10, 10), (10, 10)],
+        [
+            (10, 10, 1),
+            (
+                10,
+                11,
+            ),
+        ],
```

Comment:
Black is behaving oddly here?

Reply:
Yeah I probably had a trailing comma and it got over zealous.

Reply:
I believe different versions of black have different opinions on trailing (pseudo-Oxford?) commas.